File: solutions/028/028.go
package main

import (
	"fmt"
)

func main() {
	a := func(n int) int {
		return int(n*(n+2)/4) + int((n%4)/3) + 1
	}
	n := 1001
	sum := 0
	for i := 1; a(i) <= n*n; i++ {
		sum += a(i)
	}
	fmt.Println(sum)
}

File: solutions/028/_028.cs
using System;

public class _028 {
    public static void Main(string[] args) {
        int n = 1001;
        int sum = 0;
        for (int i = 1; a(i) <= n*n; i++)
            sum += a(i);
        Console.WriteLine(sum);
    }

    private static int a(int n) {
        return n*(n+2)/4 + (n%4)/3 + 1;
    }
}
